lib/administration/ticket/include/irods/ticket_administration.hpp

```diff
@@ -0,0 +1,577 @@
+        ///
+        /// \since 4.3.1
+        template <typename TicketConstraint>
+        void add_ticket_constraint(RxComm& conn, const int ticket_id, const TicketConstraint& constraint)
+        {
+            add_ticket_constraint(conn, std::to_string(ticket_id), constraint);
+        }
+
+        /// \brief Add the ticket constraint to the specified ticket using admin privilege
+        ///
+        /// \tparam TicketConstraint The template struct to indicate the type of constraint (only user, group, and host constraint struct) 
+        /// \param[in] admin_tag Struct tag that indicated admin privilege
+        /// \param[in] conn The communication object
+        /// \param[in] ticket_id The id of the ticket
+        /// \param[in] constraint The constraint that should be added to the ticket
+        ///
+        /// \since 4.3.1
+        template <typename TicketConstraint>
+        void add_ticket_constraint(admin_tag,
+                                   RxComm& conn,
+                                   const std::string_view ticket_name,
+                                   const TicketConstraint& constraint)
+        {
+            if constexpr (std::is_same_v<user_constraint, TicketConstraint>) {
+                detail::execute_ticket_operation(conn, "mod", ticket_name, "add", "user", constraint.value, "", true);
+            }
+            else if constexpr (std::is_same_v<group_constraint, TicketConstraint>) {
+                detail::execute_ticket_operation(conn, "mod", ticket_name, "add", "group", constraint.value, "", true);
+            }
+            else if constexpr (std::is_same_v<host_constraint, TicketConstraint>) {
+                detail::execute_ticket_operation(conn, "mod", ticket_name, "add", "host", constraint.value, "", true);
+            }
+            else {
+                throw std::invalid_argument("Invalid constraint type.");
+            }
+        }
+
+        
+        /// \brief Add the ticket constraint to the specified ticket using admin privilege
+        ///
+        /// \tparam TicketConstraint The template struct to indicate the type of constraint (only user, group, and host constraint struct) 
+        /// \param[in] admin_tag Struct tag that indicated admin privilege 
+        /// \param[in] conn The communication object 
+        /// \param[in] ticket_id The id of the ticket 
+        /// \param[in] constraint The constraint that should be added to the ticket
+        ///
+        /// \since 4.3.1
+        template <typename TicketConstraint>
+        void add_ticket_constraint(admin_tag, RxComm& conn, const int ticket_id, const TicketConstraint& constraint)
+        {
+            add_ticket_constraint(admin_tag{}, conn, std::to_string(ticket_id), constraint);
+        }
+
+        /// \brief Set constraint to the ticket specified
+        ///
+        /// \tparam TicketConstraint The template struct to indicate the type of constraint (only the ticket property constraint)
```

Comment:
Add an empty line after this line so that the `\param` directive is more noticeable.